docs/opcodes_details.md

```diff
@@ -217,6 +217,20 @@ see `CALLDATACOPY`, everything is same just the data is from the code not input
 - **evmone**: Data is in `state.host.get_tx_context().block_coinbase`
 - **openethereum**: Data is in `ext.env_info().author.clone()`
 
+## `0x46 CHAINID`
+`0x46, CHAINID, 2, base, 0, 1, Get the chain ID`
+
+- **geth**: Data is in `interpreter.evm.chainConfig.ChainID`
+- **evmone**: Data is in `state.host.get_tx_context().chain_id`
+- **openethereum**: Data is in `ext.chain_id().into()`
+
+## `0x47 SELFBALANCE`
+`0x47, SELFBALANCE, 5, low, 0, 1, Get balance of currently executing account`
+
+- **geth**: Cost the same as `BALANCE`
```

Comment:
This isn't exactly true. I think SELFBALANCE is considered "cheaper" because the lookups to the trie are expected to be cached (the address of the balance lookup doesn't change).

However, we should ask, whether we should consider this an I/O OPCODE (and exclude from scope) or not I/O (and include). My take on this is it's fine to include and observe the results, @chfast?

Reply:
`BALANCE` cost is much bigger. At the time `SELFBALANCE` was introduced the cost was `700`, currently it depends on state access cache but it should be at least `100`.

Reply:
I should have made it clearer. The intention here is to show the actual calculation cost for each implementation. And that does not seem to be much different to `BALANCE` in each case.

I.e. in geth `BALANCE` is coded as:
```go
func opBalance(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
	slot := scope.Stack.peek()
	address := common.Address(slot.Bytes20())
	slot.SetFromBig(interpreter.evm.StateDB.GetBalance(address))
	return nil, nil
}
```

and `SELFBALANCE` as:

```go
func opSelfBalance(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
	balance, _ := uint256.FromBig(interpreter.evm.StateDB.GetBalance(scope.Contract.Address()))
	scope.Stack.push(balance)
	return nil, nil
}
```

Apart from minor differences, the two methods essentially do the same thing: access the state for the balance and push the value to the stack. 

We can observe the same for evmone. `BALANCE` coded as:
```c
    x = intx::be::load<uint256>(state.host.get_balance(addr));
```
and `SELFBALANCE` as :
```c++
    state.stack.push(intx::be::load<uint256>(state.host.get_balance(state.msg->recipient)));
```

Finally for openethereum things are not different, `BALANCE`:
```rust
  let address = u256_to_address(&self.stack.pop_back());
  let balance = ext.balance(&address)?;
  self.stack.push(balance);
```
and `SELFBALANCE`:
```rust
  self.stack.push(ext.balance(&self.params.address)?);
```

As the result of this analysis, we can state that the real calculation cost of `SELFBALANCE` is similar to `BALANCE'. 

So we can either include `BALANCE` in the further time analysis and compare it to `SELFBALANCE`,  or drop both opcodes as irrelevant for this phase.

Reply:
but SELFBALANCE call refers to state which is cached/preloaded, so it shouldn't hit the I/O, and also is priced much lower. Let's include SELFBALANCE for now and exclude if it turns out problematic.